src/pool/cli.c

```diff
@@ -318,9 +318,9 @@ process_query_reply(struct dc_pool *pool, struct pool_buf *map_buf,
 		info->pi_nnodes		= map_buf->pb_node_nr;
 		info->pi_map_ver	= map_version;
 		info->pi_leader		= leader_rank;
-		if (info->pi_bits & DPI_SPACE)
+		if (connect || (info->pi_bits & DPI_SPACE))
 			info->pi_space		= *ps;
-		if (info->pi_bits & DPI_REBUILD_STATUS)
+		if (connect || (info->pi_bits & DPI_REBUILD_STATUS))
```

Comment:
Sorry, a bit confused here. could you please explain a bit? why it should (can) get rebuild status during connect?

Reply:
now the pool connect handler always returns the rebuild status and space usage, and for api usage it is not required to ask user set the pi_bits for now. 
Another option is only returns those info only when use set pi_bits for pool_connect (same as pool_query).
Either way is fine to me. 

BTW, the pi_bits is added by me a few month ago, main reason is that pool_query is needed in IO path like that refresh pool map in that case need not query rebuild status and space usage (with high bcast overhead).

Reply:
It looks to me it's better to use the pi_bits to only make the decision (on both client and server):

- If the space and rebuild status are optional to pool connect, we keep current client code unchanged, probably server code needs be adjusted to check these two bits on connect;
- If the space and rebuild status are must have for pool connect, we just set the two bits arbitrarily in dc_pool_connect().

What do you think? BTW, the commit comment section should not be empty.

Reply:
the rebuild status and space info should be optional for pool connect. I thought it is not harmful just to always returns it to user, as pool connect is not in IO critical path (but pool query is for some cases). And I did not want to require user to set pi_bits for pool connect.
Seems both you prefer to make it optional same for pool_query, I'll refresh it.